Workload-Security/Integration/aws-python-amazon-inspector-integration/c1-inspector-findings-protection.yaml

```diff
@@ -0,0 +1,837 @@
+                                    )
+                                else:
+                                    add_rules_to_computer(conf, api_version, computer.id, rule_id)
+                                    add_rules_output[instance_id] = ips_rules_info[rule_id]["name"]
+                print(f"added rules to {len(add_rules_output)} computer(s)")
+                print(add_rules_output)
+
+                if add_rules_output:
+                    # Specify a configuration set. If you do not want to use a configuration
+                    # set, comment the following variable, and the
+                    # ConfigurationSetName=CONFIGURATION_SET argument below.
+
+                    # The subject line for the email.
+                    subject = "Added Rules To Instances"
+                    # The email body for recipients with non-HTML email clients.
+                    body = f"Added rules:\r\n" "{add_rules_output}\r\n"
+
+                    added_rules_table_html = '<table border="1">' + "\n"
+                    # write headers
+                    added_rules_table_html += "</tr>" + "\n"
+                    for col in ["instance id", "rule name"]:
+                        added_rules_table_html += f"<th>{col}</th>" + "\n"
+                    added_rules_table_html += "</tr>" + "\n"
+                    # write rows
+                    for instance_id in add_rules_output:
+                        added_rules_table_html += "<tr>" + "\n"
+                        added_rules_table_html += f"<td>{instance_id}</td>" + "\n"
+                        added_rules_table_html += f"<td>{add_rules_output[instance_id]}</td>" + "\n"
+                        added_rules_table_html += "</tr>" + "\n"
+                    added_rules_table_html += "</table>"
+
+                    # The HTML body of the email.
+                    body_html = f"""<html>
+                    <head></head>
+                    <body>
+                    <h1>Added rules</h1>
+                    {added_rules_table_html}
+                    </body>
+                    </html>
+                    """
+
+                    send_email(SENDER, RECIPIENTS.split(","), subject, body, body_html)
+
+                # trigger auto ips rule assign lambda and wait for it to complete
+                lambda_client = boto3.client('lambda')
+                lambda_payload = b''
+                lambda_client.invoke(FunctionName=VULNERABILITY_REPORT_LAMBDA_FUNCTION, 
+                                        InvocationType='Event',
+                                        Payload=lambda_payload)
+                print('triggered vulnerability report lambda function')
+
+            ######################################## Uncomment the following if you running on your local machine
+            # lambda_handler(None, None)
+
+
+      FunctionName: "AutoAssignIpsRuleLambda"
+      Handler: "index.lambda_handler"
+      Layers:
+        - !Ref LambdaLayer1
```

Comment:
Might be good to have a name for the layer that gives us an idea of what the layer actually does...